### ISSUE TITLE
Plugin which replaces all image filenames with image.ext

### ISSUE THREAD

**Issue by cello-soul-termite** [Aliucord/plugin-requests#1127]
### Plugin Description

I find myself sharing images from Telegram, and their filenames contain the date they were downloaded
Replacing the filename of all images to a default would give me peace of mind

### Go into more detail...

I prefer image.ext over unknown.ext because it's the latest and doesn't reveal that I'm using an old app version. unknown also becomes Unbekannt if my language is set to German, which is another fingerprinting vector

It would be nice to have a toggle, just like the spoiler toggle, to keep the default. Or a button to rename it, but that's already out of scope
A way to pick a default for different extensions (image for png jpg jpeg gif webp etc, info for txt json csv, movie for mp4 mov, sound for mp3 flac and so on) would be welcome too

### Request Agreement

- [X] I have made sure my plugin is possible and abides by the rules!

### ​

- [X] I did indeed check to make sure my plugin request is original!

**Comment by cello-soul-termite** [comment]
placing a $15 bounty on this plugin in any cryptocurrency since idfk how to make plugins :(

**Comment by RentzNato** [comment]
Oi

**Comment by Joeeeeb5** [comment]
can someone make this

**Comment by cello-soul-termite** [comment]
vencord has this, why can't we
~bump~

**Comment by jedenastka** [comment]
Since I already revealed that I'm working on this, doesn't hurt to say it here as well.

I also made a test build available on the Aliucord server: [message link](https://discord.com/channels/811255666990907402/811261298997460992/1212201545039478804). It only includes basic functionality - simply renames all uploads to `file.[ext]`.

The full initial version will include a rule-based system to allow renaming according to file extensions and regex.

Future versions will also include a per-upload toggle and a way to manually rename files, which should cover all the needs mentioned in this request.
